File: chat-uikit/src/main/java/io/agora/chat/uikit/chathistory/widget/EaseChatRowHistoryCombine.java
package io.agora.chat.uikit.chathistory.widget;

import android.content.Context;
import android.view.View;
import android.widget.TextView;

import java.util.Date;

import io.agora.chat.ChatMessage;
import io.agora.chat.uikit.R;
import io.agora.chat.uikit.utils.EaseDateUtils;
import io.agora.chat.uikit.widget.chatrow.EaseChatRowCombine;


public class EaseChatRowHistoryCombine extends EaseChatRowCombine {

    public EaseChatRowHistoryCombine(Context context, boolean isSender) {
		super(context, isSender);
	}

    public EaseChatRowHistoryCombine(Context context, ChatMessage message, int position, Object adapter) {
		super(context, message, position, adapter);
	}

	@Override
	protected void onInflateView() {
		inflater.inflate(R.layout.ease_row_history_combine, this);
	}

	@Override
	public void setOtherTimestamp(ChatMessage preMessage) {
		timeStampView.setText(EaseDateUtils.getTimestampString(getContext(), new Date(message.getMsgTime())));
		timeStampView.setVisibility(View.VISIBLE);
	}
}

File: ease-im-kit/src/main/kotlin/io/agora/uikit/feature/chat/chathistory/widget/EaseChatRowHistoryCombine.kt
package io.agora.uikit.feature.chat.chathistory.widget

import android.content.Context
import android.util.AttributeSet
import io.agora.uikit.R
import io.agora.uikit.common.ChatMessage
import io.agora.uikit.common.extensions.getDateFormat
import io.agora.uikit.widget.chatrow.EaseChatRowCombine

open class EaseChatRowHistoryCombine @JvmOverloads constructor(
    private val context: Context,
    private val attrs: AttributeSet? = null,
    private val defStyleAttr: Int = 0,
    isSender: Boolean = false
): EaseChatRowCombine(context, attrs, defStyleAttr, isSender) {

    override fun onInflateView() {
        inflater.inflate(R.layout.ease_row_history_combine, this)
    }

    override fun onSetUpView() {
        contentView?.text = context.getString(R.string.ease_combine)
        usernickView?.let {
            if (!it.text.toString().trim().isNullOrEmpty()) {
                it.visibility = VISIBLE
            }
        }
    }

    override fun setOtherTimestamp(preMessage: ChatMessage?) {
        timeStampView?.let {
            preMessage?.let { msg ->
                it.text = msg.getDateFormat(true)
                it.visibility = VISIBLE
            }
        }
    }
}
